SDK with SpringBoot

EricksonLOOP opened the pull request:
Este pull esta a fazer uma mudança ao SDK. A mudança consiste na adição do SpringBoot Framework e melhorias de mantimento de código complementando para a facilidade e capacidade de integração com o SDK do AbacatePay Gateway JAVA.
Mudanças:

- Arquitetura Modular:
As ações foram separadas para que possar ser possível mudanças e adição de features sem o prejudicamento do código padrão. Melhora a organização, manutenção, teste, reutilização e escalabilidade do SDK, tornando o código mais limpo e fácil de gerenciar.

- Remoção do Feing:
 Feing foi removido e trocado para o OkHttp, pois assim é possível ter mais controlles sobre as requisições HTTP, lidando diretamente com a construção das requests e responses.

- Adição do SpringBoot:
 O SpringBoot foi adicionado para garantir a facilidade de configuração com o auto-configuration o que facilita o setup de bibliotecas e frameworks, injeção de dependências com o @Autowired o que ajuda no gerenciamento de objetos e classes. Adiciona um ambiente testavel com o spring-boot-starter-test para manter a qualidade do SDK. Contribui para a manutenção e contribuição e faz o complemento com suporte a Web APIs permitindo interações com REST.

- Adição do Jakarta Validation:
  O Jakarta validation será usado para a validação automática dos dados com as anotações direta nas classes como @NotNull @Email, etc. Deixar o código mais limpo e facilitar na criação de APIs Rest.

- Adição do JSON Library (org.json)
  JSON Lib ira ajudar na manipulação simple de objetos JSON e permitira uma maior facilidade na integração com o OkHttp também deixando o código mais leve e simples.

EricksonLOOP committed:
Applying Modular Architecture

Applying modular architecture for better maintenance and feature creation

EricksonLOOP committed:
feat/SeparandoServiços

Separando os serviços para melhor entendimento do código, e também para melhor manutenção e colaboração.

EricksonLOOP committed:
Add/SpringBoot Framework

Adicionando SpringBoot Framework para melhorar o mantimento das features

EricksonLOOP committed:
HotFix/SpringBoot & Feing Annotation

Adicionando anotação de inicialização do SpringBoot e Feing

EricksonLOOP committed:
Add & Remove/ Validation - Feing

Adicionando Jakarta validation para validação de entrada nos inputs e removendo pre-configurações do feing no SDK com SpringBoot

EricksonLOOP committed:
Delete AbacatePay.java

Deletando AbacatePay.java para configuração do SpringBoot como main runner

EricksonLOOP committed:
Separando módulos

Separando módulos para ter serviços separados

EricksonLOOP committed:
Created AbacatePayClient requeste & response

Criação dos modelos de Requisição e Responsta de um cliente do AbacatePay

EricksonLOOP committed:
Created & Add - AbacatePayConfig & OkHttp

Criação do AbacatePayConfig para inicializar o AbacatePay e adição do OkHttp para requisições.

EricksonLOOP committed:
Created | Feat - JsonUtilService | Criar AbacatepayClient

Agora é possível criar um novo usuário a partir de um método no serviço de cliente no AbacatePayCliente. Agora tambem é possível trabalhar com json de forma rápida com a lib JSON no JsonUtilService

EricksonLOOP committed:
Remoção Feing | Mudança Customer | Adição ParseJson

Remoção do Feing para trabalhar somente com OkHttp e JSONObject, Mudança de Customer para AbacatePayCliente -> Agora se tem dois modelos um modelo de requisição "AbacatePayClientRequest" e um de resposta "AbacatePayClientResponse". Também houve a adição de chamadas diretas pelo SDK para agilização ao desenvolver com o SDK JAVA com SpringBoot.

EricksonLOOP committed:
Added | Changed - resources | Configuration

Foi adicionando a pasta resources para configuração da aplicação através de YAML e mudança da configuração padrão de uma classe para YAML

EricksonLOOP committed:
HotFix/ Lombok dependency + Config Getters

Foi arrumado a versão do lombok e arrumado os getters no clientservice

EricksonLOOP committed:
feat/ CreateBilling

Agora é possível criar cobranças através do métdo create billing no ABacatePayServices

EricksonLOOP committed:
Fix | Feat - FIxed Builder Error | Feat Tests

Erro ao rodar projeto no builder do CreateBilling mais criação de testes

EricksonLOOP committed:
Delete TesteFiles

Deletando os arquivos de testes para uma escritura de testes mais centralizados.

EricksonLOOP committed:
Created JAVA JAR - Delete Testes

Como o projeto ainda está minusculo é mais facil testar tudo na mão

EricksonLOOP committed:
updt Readme.md

EricksonLOOP committed:
updt Java

From 11 to 17

EricksonLOOP committed:
Add JitPack

EricksonLOOP committed:
Removed SpringBoot

Springboot como gereniador principal removido

ssimiao commented on pom.xml:
```diff
@@ -6,96 +6,83 @@
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-core</artifactId>
-            <version>3.11.2</version>
+            <groupId>org.springframework.boot</groupId>
```
ao adicionar o spring, você está limitando o uso do sdk apenas a esse framework, não? se alguém quiser usar ele com quarkus, micronaut, rxjava ele vai ter dificuldades, entendo que a ideia da criação de um sdk ainda mais para uma api simples, seria facilitar a vida da pessoa que vai utilizar ele.

EricksonLOOP replied:
A ideia de adicionar uma versão do SDK com Spring Boot é justamente facilitar a integração com aplicações que já utilizam esse framework. A grande vantagem é que, ao consumir o SDK em projetos com Spring Boot, o próprio framework pode gerenciar as classes e serviços do SDK automaticamente, sem a necessidade de configuração manual. Isso otimiza o processo de desenvolvimento, reduz a duplicação de código e facilita a manutenção, especialmente para regras de negócio e serviços já definidos no SDK. Essa abordagem também facilita a transição para Gateways de Pagamento que utilizam Spring Boot, tornando o processo mais ágil.

Uma versão desacoplada do SDK (sem dependências do Spring) já está em desenvolvimento e garantirá maior flexibilidade, permitindo que o SDK seja facilmente integrado a diferentes ambientes, enquanto mantém a facilidade de uso para quem utiliza Spring Boot.

Vejo um grande potencial de crescimento para o projeto AbacatePay, e por isso, estou criando uma base sólida agora, preparando o SDK para quando for necessário integrá-lo com SpringBoot futuramente.

EricksonLOOP committed:
Changed Main Method

Main methdo without SpringbootApllication anotation

EricksonLOOP committed:
TryingTofix Dependence Error

EricksonLOOP committed:
Changed: addClassPath false

Ao gerar um arquivo modular para java com as dependências do SpringBoot, esta ocorrendo um erro muito chato das classes não estarem sendo executadas no contexto spring. Pois ao gerar o módulo estava vindo um jar executavel é não um jar modulo (lib).

EricksonLOOP committed:
NoPath

EricksonLOOP committed:
HotFix/AbacatePayConfig

Arrumado a feature de configurar a BASE_URL e a API_KEY através do application.properties ou .yaml/.yml

Arrumado o parâmetro de requisição na URL.

Bug: Falta de MetaData na resposta JSON da API do AbacatePay - Em análise

EricksonLOOP committed:
Fixed/"Metadata is missing in Response"

O erro foi arrumado, era de fato que a resposta vinha como:

    "error": null,
    "data": {
        "id": "cust_r5EzqgfrWTtnHqhun3qyfAwu",
        "metadata": {
            "name": "Seu zé Da silva",
            "cellphone": "+99999999999",
            "taxId": "1234123789",
            "email": "ze.silva@exampledois.com"
        }
    }
}

Mas no JsonUtil esperava de outra forma.

EricksonLOOP committed:
fixed/"Metadata is missing in response"

EricksonLOOP committed:
Fixed/Array must start with '[' -> error

Não estava conseguindo acessar o indicie específico do array para montar os objetos, pois não estava sendo reconhecido como array. Erro consertado.

EricksonLOOP committed:
fixed/AbacatePayClientRequest

O modelo de requisição agora possui os parâmetros corretos.

EricksonLOOP committed:
fixed/No default constructor for CreateBillingProduct

Para o fix, foi necessário remover o @Builder já que não tinha uso nenhum, e criar um construtor novo sem parametros.

EricksonLOOP committed:
fixed/createBillingResponseFromJsonToObject

Agora é possível converter a resposta em JSON da api AbacatePay para objeto java

EricksonLOOP committed:
update jsonutil

EricksonLOOP committed:
Create setting.json

EricksonLOOP committed:
Add @Component at config
